downloads/swan-lake-release-notes/swan-lake-alpha/RELEASE_NOTE.md

```diff
@@ -0,0 +1,1432 @@
+match v {
+    error ("Message") => {
+    // Match errors with "Message" as the error message.
+    }error () => {
+    // Match all errors.
+    }}
+```
+
+###### Improved Type Narrowing Within Match Statements
+
+When code is executed through each match-clause, the type of the matched expression is narrowed. In the example below, the type of `v` in the last match clause is narrowed to `string`.
+
+```ballerina
+match v {
+    var a if a is int|boolean => {
+        return a.toString();
+    }
+    _ => {
+        return v; // Type of `v` is `string` here.
+    }
+}
+```
+
+##### Support for Cyclic Union Types
+
+Cyclic union types are now supported. A cyclic union type descriptor can directly refer to its identifier in its type descriptor.
+
+```ballerina
+type Integers int|Integers[]|map<Integers>;
+
+Integers intValue = 5;
+Integers intArray = [1, 2, 3, 4];
+Integers intMap = {i: 1, j: 2};
+Integers integers = [intValue, intArray, intMap];
+```
+
+##### Updated Syntax for User-Defined Error Constructions
+
+The error constructor expression now requires the `error` keyword to construct a user-defined error.
+
+**Previous Syntax:**
+```ballerina
+MyError myError = MyError("Message");
+```
+ 
+**New Syntax:**
+```ballerina
+MyError myError = error MyError("Message");
+```
+
+#### Changes on Casting with Errors
+
+Errors cannot be cast away (i.e., if the value that is being cast can be an error, the type to which the cast is attempted should also have a subtype of error). 
+
+The following was allowed previously.
+
+```ballerina
+function foo() returns string|int|error {
+    return error(“Error Message”);
+}
+
+public function main() {
+    string s = <string>foo(); // Cast without considering the error.
+}
+```
+
+This is now disallowed and can be rewritten as follows.
+
+```ballerina
+function foo() returns string|int|error {
+    return error("Error Message");
+}
+
+public function main() {
+    string s = <string>checkpanic foo(); // use`checkpanic` to panic if `foo()` returns an error and then attempt the cast
+}
+```
+
+1. Using `checkpanic` - This will preserve the `panic` behavior (but with a different error - the previous error was a cast error. Now, it will be the actual error returned by `foo()`).
+
+    ```ballerina
+    function foo() returns string|int|error {
+        return error("Error Message");
+    }
+
+    public function main() {
+        string s = <string>checkpanic foo(); // use`checkpanic` to panic if `foo()` returns an error and then attempt the cast
+    }
+    ```
+
+2. Using `check` - Alternatively, if the function’s return type allows returning the error, `check` can be used before attempting the cast to return the error instead of panicking.
+
+```ballerina
+function foo() returns string|int|error {
+    return error(“Error Message”);
+}
+
+public function main() returns error? {
+    string s = <string>check foo(); // use`check` to return the error if `foo()` returns an error, and then attempt the cast
+}
+```
```

Comment:
```suggestion
    ```ballerina
    function foo() returns string|int|error {
        return error(“Error Message”);
    }

    public function main() returns error? {
        // Use`check` to return the error if `foo()` returns an error, and then attempt the cast.
        string s = <string>check foo();
    }
    ```
```